src/main/java/randoop/contract/ObserverEqArray.java

```diff
@@ -0,0 +1,205 @@
+  }
+
+  @Override
+  public String toCodeString() {
+    if (value.getClass().getComponentType() == float.class
+        || value.getClass().getComponentType() == double.class) {
+      return String.format(
+          "org.junit.Assert.assertArrayEquals(x0, %s, %s);", printArray(), FLOATING_POINT_DELTA);
+    } else {
+      return String.format("org.junit.Assert.assertArrayEquals(x0, %s);", printArray());
+    }
+  }
+
+  /**
+   * Prints the code string of the second parameter (instantiation of array) in assertArrayEquals
+   * (e.g. prints the second parameter in assertArrayEquals(var, new int[] {1,2,3})).
+   *
+   * @return String that represents an instantiation of an array equal to value
+   */
+  private String printArray() {
+    return "new " + value.getClass().getCanonicalName() + printArrayComponents();
+  }
+
+  /**
+   * Converts an Object into its array representation.
+   *
+   * @param value the given Object
+   * @return its array representation
+   */
+  private Object[] toObjectArray(Object value) {
+    if (value instanceof byte[]) {
+      return ArrayUtils.toObject((byte[]) value);
+    } else if (value instanceof short[]) {
+      return ArrayUtils.toObject((short[]) value);
+    } else if (value instanceof int[]) {
+      return ArrayUtils.toObject((int[]) value);
+    } else if (value instanceof long[]) {
+      return ArrayUtils.toObject((long[]) value);
+    } else if (value instanceof float[]) {
+      return ArrayUtils.toObject((float[]) value);
+    } else if (value instanceof double[]) {
+      return ArrayUtils.toObject((double[]) value);
+    } else if (value instanceof char[]) {
+      return ArrayUtils.toObject((char[]) value);
+    } else if (value instanceof boolean[]) {
+      return ArrayUtils.toObject((boolean[]) value);
+    } else {
+      return (Object[]) value;
+    }
+  }
+
+  /**
+   * Returns a string representation of the components of the array, enclosed in curly braces, as it
+   * would appear in Java source code.
+   *
+   * @return a String that represents the components of the array
+   */
+  private String printArrayComponents() {
+    StringJoiner sj = new StringJoiner(", ", "{", "}");
+    int length = Array.getLength(value);
+    for (int i = 0; i < length; i++) {
+      sj.add(printLiteralValue(Array.get(value, i)));
+    }
+    return sj.toString();
+  }
+
+  /**
+   * Prints one literal value, as it would appear in Java code
+   *
+   * @return the Java code for the literal value
+   */
+  private String printLiteralValue(Object element) {
+    if (element == null) {
+      return "null";
+    } else if (element.getClass().isEnum()) {
+      return new EnumValue((Enum<?>) element).getValueName();
+    } else if (element.getClass() == Class.class) {
+      return ((Class<?>) element).getCanonicalName() + ".class";
+    } else {
+      return Value.toCodeString(element);
+    }
+  }
+
+  @Override
+  public boolean evaluate(Object... objects) throws Throwable {
```

Comment:
This method should return true or false, rather than throwing an exception.

Reply:
Randoop fails if I try to surround these statements in a try-catch block because it warns that I am ignoring an AssertionError. How should I get around this?

Reply:
I just added a @SuppressWarnings("AssertionFailureIgnored") tag. Let me know if you think this is the wrong fix, or there is a better way.

Reply:
Always write a justifying comment on any `@SuppressWarnings` annotation.  (I did this.)